tsl/src/continuous_aggs/materialize.c

```diff
@@ -197,8 +198,21 @@ continuous_agg_materialize(int32 materialization_id, ContinuousAggMatOptions *op
 
 	catalog = ts_catalog_get();
 
-	/* copy over all the materializations from the raw hypertable to all the continuous aggs */
+	/* copy over all the materializations from the raw hypertable to all the continuous aggs
+	 */
 	caggs = ts_continuous_aggs_find_by_raw_table_id(cagg_data.raw_hypertable_id);
+	if (list_length(caggs) > 1)
```

Comment:
Isn't it more safe to always lock here? Could there be a case when cagg_2 is created in another transaction, but not yet visible to this transaction? If cagg_2 is immediately refreshed, I imagine that you won't lock in that case, although you still have concurrent refreshes.